build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java

```diff
@@ -561,7 +561,9 @@ private void updateBuildTaskStatus(BuildTask task, BuildCoordinationStatus statu
 
         userLog.info("Build status updated to {}; previous: {}", status, oldStatus);
 
-        if (BuildStatus.fromBuildCoordinationStatus(oldStatus) != BuildStatus.fromBuildCoordinationStatus(status)) {
+        BuildStatus oldBuildStatus = BuildStatus.fromBuildCoordinationStatus(oldStatus);
+        BuildStatus newBuildStatus = BuildStatus.fromBuildCoordinationStatus(status);
+        if ((oldBuildStatus != newBuildStatus) && !(oldBuildStatus.isFinal() && newBuildStatus.isFinal())) {
```

Comment:
Can you merge this with the condition above L543?

Reply:
Do you mean that I should use the same condition also on L543?

Reply:
I've added the edge-case to markFinished condition also with some clarifying comments.